### DISCUSSION TITLE
Why topic-level policies are not supported for non-persistent topics?

### DISCUSSION THREAD

**Discussion by visortelle** [apache/pulsar#18346] [Q&A] [answered]
I believe, that some subset of policies would be useful for non-persistent topics.

But I see `validatePersistentTopic()` for all topic-level policies.

https://github.com/apache/pulsar/blob/82237d3684fe506bcb6426b3b23f413422e6e4fb/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java#L951

Are topic-level policies aren't just implemented yet for non-persistent topics?

Thank you.

**Comment by michaeljmarshall** [comment, marked as answer]
I am pretty sure they are only available for persistent topics at this time. Which policies do you think we should expose for non-persistent topics?

**Comment by visortelle** [comment]
I think that all namespace-level policies which affect non-persistent topics should also be configurable at the topic-level.

I suppose that's probably all, except persistence and geo-replication related.

I personally don't have any specific needs for it at this time, just asking.
